src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -34,139 +37,168 @@ public class ErrorListService
             }
             else
             {
                 // They're opening a non-SARIF log, so we need to convert it.
                 // Ask if they'd like to save the converted log.
-                MessageDialogCommand response = promptOnLogConversions ? PromptToSaveProcessedLog(Resources.ConvertNonSarifLog_DialogMessage) : MessageDialogCommand.No;
+                MessageDialogCommand response = promptOnLogConversions ?
+                    await PromptToSaveProcessedLogAsync(Resources.ConvertNonSarifLog_DialogMessage).ConfigureAwait(continueOnCapturedContext: false) :
+                    MessageDialogCommand.No;
 
                 if (response == MessageDialogCommand.Cancel)
                 {
                     return;
                 }
 
-                var converter = new ToolFormatConverter();
-                var sb = new StringBuilder();
-
-                using (var input = new MemoryStream(File.ReadAllBytes(filePath)))
-                {
-                    var outputTextWriter = new StringWriter(sb);                
-                    var outputJson = new JsonTextWriter(outputTextWriter);
-                    var output = new ResultLogJsonWriter(outputJson);
-
-                    input.Seek(0, SeekOrigin.Begin);
-                    converter.ConvertToStandardFormat(toolFormat, input, output);
-
-                    // This is serving as a flush mechanism.
-                    output.Dispose();
+                // The converter doesn't have async methods, so spin
+                // up a task to do this.
+                await System.Threading.Tasks.Task.Run(() => {
+                    var sb = new StringBuilder();
+                    using (FileStream fileStream = File.OpenRead(filePath))
+                    {
+                        using (var outputTextWriter = new StringWriter(sb))
+                        using (var outputJson = new JsonTextWriter(outputTextWriter))
+                        using (var output = new ResultLogJsonWriter(outputJson))
+                        {
+                            fileStream.Seek(0, SeekOrigin.Begin);
+                            var converter = new ToolFormatConverter();
+                            converter.ConvertToStandardFormat(toolFormat, fileStream, output);
+                        }
 
-                    logText = sb.ToString();
+                        logText = sb.ToString();
 
-                    if (response == MessageDialogCommand.Yes)
-                    {
-                        // Prompt for a location to save the converted log.
-                        outputPath = PromptForFileSaveLocation(Resources.SaveConvertedLog_DialogTitle, filePath);
+                        if (response == MessageDialogCommand.Yes)
+                        {
+                            // Prompt for a location to save the converted log.
+                            outputPath = PromptForFileSaveLocationAsync(Resources.SaveConvertedLog_DialogTitle, filePath).Result;
```

Comment:
How does this work as you are on some random thread inside a task?

Reply:
You answered your own question. Because we are already on another thread, we do not need to spin up yet another thread (i.e. call await).

---
In reply to: [494563115](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494563115) [](ancestors = 494563115)